modules/rostests/apitests/win32nt/ntuser/NtUserSetTimer.c

```diff
@@ -1,18 +1,247 @@
 /*
- * PROJECT:         ReactOS api tests
- * LICENSE:         GPL - See COPYING in the top level directory
- * PURPOSE:         Test for NtUserSetTimer
- * PROGRAMMERS:
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Tests for NtUserSetTimer
+ * COPYRIGHT:   Copyright 2008 Timo Kreuzer <timo.kreuzer@reactos.org>
+ *              Copyright 2024 Tomáš Veselý <turican0@gmail.com>
  */
 
 #include "../win32nt.h"
 
+#define SLEEP_TIME 500
+#define TIME_TOLERANCE 0.5
+
+#define TEST1_COUNT 20
+#define TEST1_INTERVAL 10
+
+#define TEST2_COUNT 40000
+#define TEST2_INTERVAL 10
+
+#define TESTW1_COUNT 20
+#define TESTW1_INTERVAL 10
+
+#define TESTW2_COUNT 40000
+#define TESTW2_INTERVAL 10
+
+typedef struct TIMER_MESSAGE_STATE1
+{
+    UINT_PTR index;
+    UINT counter;
+} TIMER_MESSAGE_STATE1;
+
+TIMER_MESSAGE_STATE1 timerId1[TEST1_COUNT];
+
+typedef struct TIMER_MESSAGE_STATEW1
+{
+    UINT counter;
+} TIMER_MESSAGE_STATEW1;
+
+TIMER_MESSAGE_STATEW1 timerIdW1[TESTW1_COUNT];
+
+/* TIMERPROC for the test1,2,3() with messages without window */
+static void CALLBACK
+TimerProc(HWND hWnd, UINT uMsg, UINT_PTR idEvent, DWORD dwTime)
+{
+    UINT i;
+    for (i = 0; i < TEST1_COUNT; i++)
+    {
+        if (timerId1[i].index == idEvent)
+            timerId1[i].counter++;
+    }
+}
+
+/* TIMERPROC for the testW1,2() with messages with window */
+static LRESULT CALLBACK
+WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    switch (uMsg)
+    {
+    case WM_TIMER:
+        timerIdW1[wParam].counter++;
+        return 0;
+    }
+
+    return DefWindowProc(hwnd, uMsg, wParam, lParam);
+}
+
+// TEST WITH MESSAGES WITHOUT WINDOW - test count of sent messages
+static BOOL test1(void)
+{
+    UINT i, countErrors = 0;
+
+    int minMessages = ((float)SLEEP_TIME / (float)TEST1_INTERVAL) * (1 - TIME_TOLERANCE);
+    int maxMessages = ((float)SLEEP_TIME / (float)TEST1_INTERVAL) * (1 + TIME_TOLERANCE);
+
+    ZeroMemory(timerId1, sizeof(timerId1));
+
+    for (i = 0; i < TEST1_COUNT; i++)
+    {
+        timerId1[i].index = SetTimer(NULL, 0, TEST1_INTERVAL, TimerProc);
+        if (timerId1[i].index == 0)
+            countErrors++;
+    }
+
+    ULONGLONG startTime = GetTickCount();
+
+    MSG msg = { NULL };
```

Comment:
Optional: If you moved their declaration to the funcs top, it would compile with more compilers. Same for test2,test3,testW1,START_TEST

Reply:
Ok, I move declarations to begin of functons.